### PR TITLE
Add: contributing md file

### PR DESCRIPTION
**Pull request by guacamole**
### Motivation & Context:
This PR is to add contribution guidelines

### Description:
With contribution guidelines in place, there will be a proper structure to contributions made to OpenRegistry and it will also make it easier to review them

### Types of Changes:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### PR Checklist:

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I've read the CONTRIBUTION guide
- [x] I have signed-off my commits with git commit -s

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Signed-off-by: <A full name> <A email>
+Co-authored-by: <B full name> <B email>
+
+```
+- Always make sure to sign your commits
+- Document your pull requests. Include the reasoning behind each change, and the testing done.
+- When raising an issue, follow the issue template in place to help us understand the issue at it's very core
+
+### Subject Line Standard Terminology
+
```
Comment (jay-dee7) [inline CONTRIBUTING.md:36]: @guacamole Can we please add the link to conventional commits site?
https://www.conventionalcommits.org/en/v1.0.0/

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+First Word | Meaning
+--- | --
```
Comment (jay-dee7) [inline CONTRIBUTING.md:38]: Also, let's stick to commits starting with lowercase letter? like:
`add: XYZ feature`
`refactor: Complex piece of code`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Fix | Fix an issue e.g. bug, typo, accident, misstatement.
+Bump | Increase the version of something e.g. dependency.
+Make | Change the build process, or tooling, or infra.
+Start | Begin doing something; e.g. create a feature flag.
+Stop | End doing something; e.g. remove a feature flag.
+Refactor | A code change that MUST be just a refactoring.
+Reformat | Refactor of formatting, e.g. omit whitespace.
+Optimize | Refactor of performance, e.g. speed up code.
+Document | Refactor of documentation, e.g. help files.
+
```
Comment (jay-dee7) [inline CONTRIBUTING.md:50]: Another addition we can do is add `!` at the end of the commit type for breaking changes? eg: There's a feature which cause a large refactor/rewrite or causes us to change other functionality, then we add the `!` mark with the commit type, declaring a breaking change:

`feat!: Github apps integration`

File: CONTRIBUTING.md (hunk 2 above)
Comment (guacamole) [reply CONTRIBUTING.md]: you got it!

**Review by jay-dee7 — APPROVED** [review]
LGTM 🥇 🥇 🥇

**Pull request merged**